core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -634,6 +636,10 @@ boolean isPreserveAliases() {
                 if (!noIndicesAllowedOrMatches(options, matches)) {
                     throw infe(expression);
                 }
+
+                if (expression.contains("*")) {
```

Comment:
> what happens with `+test1, +test2, +test3,-test2`

That would resolve to: `["test1", "test2", "test3", "-test2"]`, assuming a `-test2` index exists, which I think is the correct behavior?

If `-test2` doesn't exist then it won't be negated and in that case I'm not sure what we want to do, do you think we should support this (adding and then removing an index without wildcards)?